security/pkg/nodeagent/sds/adapter.go

```diff
@@ -32,15 +32,15 @@ type sdsServiceAdapter struct {
 }
 
 func (d *sdsServiceAdapter) FetchSecrets(ctx context.Context, request *xdsapiv2.DiscoveryRequest) (*xdsapiv2.DiscoveryResponse, error) {
-	v3Resp, err := d.s.FetchSecrets(ctx, v2.UpgradeV2Request(request))
-	return v2.DowngradeV3Response(v3Resp), err
+	v3Resp, err := d.s.FetchSecrets(ctx, xds.UpgradeV2Request(request))
```

Comment:
Here the PR changes "v2.UpgradeV2Request" to "xds.UpgradeV2Request". UpgradeV2Request is specific to "istio.io/istio/pilot/pkg/proxy/envoy/v2" but "xds.UpgradeV2Request" does not contain the version. Can you explain how the version is represented in "xds.UpgradeV2Request"?

Reply:
`UpgradeV2Request` upgrades from a request of v2 type to v3.  Moving to xds does not make any difference in that behaviour.  In subsequent PRs I plan to move this to v3 package under xds so that it is more clear. For now, I just moved every thing under v2 -> xds.

Reply:
> but "xds.UpgradeV2Request" does not contain the version

It does, that is what the "V2" is. `v2.UpgradeV2Request` is already redundant.